src/scripting/vmthunks.cpp

```diff
@@ -2503,12 +2504,146 @@ DEFINE_ACTION_FUNCTION(_Console, MidPrint)
 	return 0;
 }
 
+//==========================================================================
+//
+//
+//
+//==========================================================================
+
+static int isValid( level_info_t *info )
+{
+	return info->isValid();
+}
+
+DEFINE_ACTION_FUNCTION_NATIVE(_LevelInfo, isValid, isValid)
+{
+	PARAM_SELF_STRUCT_PROLOGUE(level_info_t);
+	ACTION_RETURN_BOOL(self->isValid());
+}
+
+static void ZLookupLevelName( level_info_t *info, FString *result )
+{
+	*result = info->LookupLevelName();
+}
+
+DEFINE_ACTION_FUNCTION_NATIVE(_LevelInfo, LookupLevelName, ZLookupLevelName)
+{
+	PARAM_SELF_STRUCT_PROLOGUE(level_info_t);
+	ACTION_RETURN_STRING(self->LookupLevelName());
+}
+
+static int GetLevelInfoCount()
+{
+	return wadlevelinfos.Size();
+}
+
+static level_info_t* GetLevelInfo( int index )
+{
+	return &wadlevelinfos[index];
+}
+
+DEFINE_ACTION_FUNCTION_NATIVE(_LevelInfo, GetLevelInfoCount, GetLevelInfoCount)
+{
+	PARAM_PROLOGUE;
+	ACTION_RETURN_INT(wadlevelinfos.Size());
+}
+
+DEFINE_ACTION_FUNCTION_NATIVE(_LevelInfo, GetLevelInfo, GetLevelInfo)
+{
+	PARAM_PROLOGUE;
+	PARAM_INT(index);
+	ACTION_RETURN_POINTER(&wadlevelinfos[index]);
```

Comment:
The same problem as with the global `GetLevelInfo()` function.

In general, it would be nice to call one function from another instead of copying its implementation. This applies to some functions as well.